packages/halo-nonce-pcd/src/HaLoNoncePCD.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  BooleanArgument,
+  DisplayOptions,
+  PCD,
+  PCDPackage,
+  SerializedPCD,
+  StringArgument,
+} from "@pcd/pcd-types";
+import { ec } from "elliptic";
+import { sha256 } from "js-sha256";
+import { v4 as uuid } from "uuid";
+import { HaLoNonceCardBody } from "./CardBody";
+
+const secp256k1 = new ec("secp256k1");
+
+export const HaLoNoncePCDTypeName = "halo-nonce-pcd";
+
+export interface HaLoNoncePCDArgs {
+  /**
+   * True if we request a new HaLoNoncePCD, false if we read from remaining params
+   */
+  requestNewPCD: BooleanArgument;
+
+  /**
+   * The uncompressed hex string of the public key on slot 2
+   */
+  pk2: StringArgument;
```

Comment:
hard to know what `slot 2` is - can you add a preamble to this file which links to relevant docs?